lib/libc/minimal/source/bsearch.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <bsearch.h>
+
+/*
+ * Generic implementation of Binary Search 
+ * @brief: binary search in an array of elements
+ * @base: pointer to first element to search
+ * @key: pointer to item being searched for
+ * @num: number of elements
+ * @size: size of each element
+ * @cmp: pointer to comparison function
+ *
+ * @return: pointer to key if present, NULL otherwise
+ *
+ * This function does a binary search on an ascending sorted order
+ * using the provided comparison function.
+ */
+
```

Comment:
These doxygen comments should be reformatted to match normal doxygen naming conventions. All of the doxygen tag available are shown here: http://www.doxygen.nl/manual/commands.html

See, for example, the [`\param` flag](http://www.doxygen.nl/manual/commands.html#cmdparam) (which can also be referenced via `@param`).

Here is an example, which doesn't need to be copied scrupulously, just something I copied and pasted from some code I'm working on right now:

```
/**
 * @brief Sets a single value at the specified row (i) and column (j).
 *
 * @param m     Pointer to the zsl_mtx to use.
 * @param i     The row number to update (0-based).
 * @param j     The column number to update (0-based).
 * @param x     The value to assign.
 *
 * @return  0 if everything executed correctly, or -EINVAL on an out of
 *          bounds error.
 */
int zsl_mtx_set(struct zsl_mtx *m, size_t i, size_t j, zsl_real_t x);
```